doc/MiniLanguageTutorial.md

```diff
@@ -480,37 +563,278 @@ Mini never automatically converts types to make an operation succeed.  Programme
+1. `Tuple(`...`)`: up to 8 AVM values.
+
+More complex values are usually encoded by storing multiple values inside a `Tuple`, included nested tuples.
+
+## Default Value
+
+Every type has a default value that is used for the initial value of globals with that type.
+The default values for each type are as follows:
+
+* `uint`, `int`, `bytes32`, `address`:  
+  `0`, represented in `AVM` as an `Int`.
+* `bool`:  
+  `false`, this is also a `0` `Int` in the underlying AVM.
+* `buffer`:  
+  A buffer with length `0`, attempting to read any bytes past the end of a buffer will be treated as reading `0` values.
+* `any`:  
+  A length `0` tuple, `()`.
+* `union`:  
+  The default value of a union type is the default value of its leftmost variant.
+* `tuple`:  
+  The default value of a tuple is a wrapping tuple, with leaves equal to the default values of each field type respectively. 
+* `struct`:  
+  The default value of a struct is a wrapping tuple, with leaves equal to the default values of each field type respectively.
+* `option`:  
+  The default value of all `option` types is the `None` variant, in AVM this is represented as a length 1 tuple containing an `Int` value `0`
+* `func`:  
+  The default value of all `func` types is a `CodePoint` 
+* `map`:  
+  An empty map, in AVM this is represented as a length 2 `Tuple` with both fields as an `Int` with value `0`.
+* `array`:  
+  An `array` of *type* has a default value of an array with length 1 containing the default value of *type*.
+  In AVM this is represented by a length 3 `Tuple` containing, `Int`s with a value of `0` for the first two fields, 
+  and a length 8 `Tuple` containing 
+* `fixed array`:  
+  A default value for a `fixed array` of type *type* and length *length* is a fixed array of length *length*, where every array element has a value of the default value of *type*.
+  See the AVM representation section for more information of how the memory for this value is layed out.
+* `nominal`, `specialized generic`:  
+  The default value for a nominal or specialized generic type is the default value of its representation.
+
+While non-specialized generic types should never be created, if they are they will have a default value of `()`.
+
+## AVM Representation
+
+Types in mini are internally represented as AVM values.
+
+1. uint:  
+   Always an `Int`, any `Int` value is allowable for this type.
+1. int:  
+   Always an `Int`, any `Int` value is allowable for this type.
+1. bool:  
+   Always an `Int`, the values `0` and `1` are allowed for this type, but nothing else.
+1. buffer:  
+   Always a `Buffer`, any value is allowable for this type.
+1. bytes32:  
+   Always an `Int`, any value is allowable for this type.
+1. address:  
+   Always an `Int`, only values up to `2^160-1` are allowed for this type, where the `Int` must be interpreted as unsigned.
+1. void:  
+   Represents lack of a value, whenever a value of type `void` is present, then it is assumed there is no value there.
```

Comment:
Can a value of type void exist?  The last part of this seems self-contradictory: when a value is present, then no value is present???

Reply:
While there is no AVM value that can be of type void, there can be values of type void created via calls to void functions or unsafecasts, the guard on let statements mean these never get assigned to a value, but they can exist within an expression.